Composer/packages/client/src/ShellApi.ts

```diff
@@ -132,8 +133,13 @@ export const ShellApi: React.FC = () => {
     if (id === undefined) throw new Error('must have a file id');
     const file = lgFiles.find(file => file.id === id);
     if (!file) throw new Error(`lg file ${id} not found`);
+    let templates: LGTemplate[] = [];
+    try {
+      templates = lgUtil.parse(file.content);
+    } catch (error) {
+      console.error(error);
```

Comment:
I am worried about data loss due to bad `lgUtil.parse`.

Reply:
here the templates is used to generate a "view" for form editor and\or other editors, the original data is not touched and can be accessed via LG all up view. 

so this `parsing` solely won't cause data loss, there is a potential loss if the upper layer (editors) think there is no templates and try to reset the whole file.  This is a typical risk, it's like you open a word document, due to bad parse the data is not loaded into your view, you choose to clean up and re-type your content.  The fix is to make sure the parse is solid.